packages/shared/src/components/modals/SteppedModal.tsx

```diff
@@ -0,0 +1,116 @@
+import classNames from 'classnames';
+import React, {
+  Children,
+  KeyboardEvent,
+  MouseEvent,
+  ReactElement,
+  useState,
+} from 'react';
+import { Button } from '../buttons/Button';
+import TabContainer, { Tab } from '../tabs/TabContainer';
+import { ModalProps, StyledModal } from './StyledModal';
+
+interface SteppedModalProps extends ModalProps {
+  isLastStepLogin?: boolean;
+  onStepChange?: (stepBefore: number, stepNow: number) => void | Promise<void>;
+  onFinish?: () => void | Promise<void>;
+}
+
+function SteppedModal({
+  children,
+  isLastStepLogin,
+  onStepChange,
+  onFinish,
+  onRequestClose,
+  ...props
+}: SteppedModalProps): ReactElement {
+  const [step, setStep] = useState(0);
+  const getWidth = () => {
+    const count = Children.count(children);
```

Comment:
Might be a bit confusing using `children` and `Children` together, but not sure if there's an alternative?

Reply:
I can probably use `React.Children` instead of importing the `Children` directly. Will refactor this one.